edge/pkg/metamanager/metaserver/server.go

```diff
@@ -45,12 +53,54 @@ func NewMetaServer() *MetaServer {
 	return &ls
 }
 
+func createTLSConfig() tls.Config {
+	ca, err := os.ReadFile(metaserverconfig.Config.TLSCaFile)
+	if err == nil {
+		block, _ := pem.Decode(ca)
+		ca = block.Bytes
+		klog.Info("Succeed in loading CA certificate from local directory")
+	}
+	pool := x509.NewCertPool()
+	ok := pool.AppendCertsFromPEM(pem.EncodeToMemory(&pem.Block{Type: certutil.CertificateBlockType, Bytes: ca}))
+	if !ok {
+		panic(fmt.Errorf("fail to load ca content"))
+	}
+	cert, err := os.ReadFile(metaserverconfig.Config.TLSCertFile)
+	if err == nil {
+		block, _ := pem.Decode(cert)
+		cert = block.Bytes
+		klog.Info("Succeed in loading certificate from local directory")
+	}
+	key, err := os.ReadFile(metaserverconfig.Config.TLSPrivateKeyFile)
+	if err == nil {
+		block, _ := pem.Decode(key)
+		key = block.Bytes
+		klog.Info("Succeed in loading private key from local directory")
+	}
+
+	certificate, err := tls.X509KeyPair(pem.EncodeToMemory(&pem.Block{Type: certutil.CertificateBlockType, Bytes: cert}), pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: key}))
+	if err != nil {
+		panic(err)
+	}
+	return tls.Config{
+		ClientCAs:    pool,
+		ClientAuth:   tls.VerifyClientCertIfGiven,
+		Certificates: []tls.Certificate{certificate},
+		MinVersion:   tls.VersionTLS12,
+	}
```

Comment:
specify  CipherSuites for security

Reply:
Insecure cipher suites will be moved after go version 1.17, there is one tracked issue.
FYI: https://github.com/kubeedge/kubeedge/pull/3924